include/chopper/layout/ibf_query_cost.hpp

```diff
@@ -58,13 +68,47 @@ class ibf_query_cost
 
 private:
 
-    // (19,19) mean c_{T_max} (see paper)
-    constexpr static inline const std::array<double, 11> cost_factor
-    {
-        //64  128   256   512  1024  2048  4096  8192   16384  32768  65536
-        {1.0, 1.1, 1.32, 1.61, 2.69, 4.48, 7.53, 13.65, 23.86, 45.66, 90.42}
+    // actual new table
+    // static inline const std::map<double, std::array<double, 11>> cost_factor
+    // {   /* FPR, cost factors relative to a 64 IBF */
+    //     {0.0001, {1.0, 0.87, 1.07, 1.58, 1.86, 2.3, 3.56, 4.78, 6.89, 11.61, 22.7}},
+    //     {0.0005, {1.0, 1.2, 1.04, 1.39, 1.91, 2.87, 3.88, 7.28, 7.99, 14.3, 28.34}},
+    //     {0.0025, {1.0, 0.9, 1.08, 1.29, 1.91, 2.46, 3.84, 5.49, 7.93, 14.17, 28.16}},
+    //     {0.0125, {1.0, 1.2, 1.36, 2.01, 2.64, 3.37, 5.98, 9.92, 16.29, 30.17, 59.08}},
+    //     {0.0625, {1.0, 0.96, 1.2, 1.71, 2.27, 3.54, 6.42, 10.69, 19.31, 37.12, 72.5}},
+    //     {0.3125, {1.0, 1.43, 1.83, 2.92, 4.68, 8.25, 16.58, 27.7, 52.3, 102.65, 202.55}}
+    // };
+
+    // dummy table to see if refactoring works before entering new values
+    static inline const std::map<double, std::array<double, 11>> cost_factor
+    {   /* FPR, cost factors relative to a 64 IBF */
+        {0.0001, {1.0, 1.1, 1.32, 1.61, 2.69, 4.48, 7.53, 13.65, 23.86, 45.66, 90.42}},
+        {0.0005, {1.0, 1.1, 1.32, 1.61, 2.69, 4.48, 7.53, 13.65, 23.86, 45.66, 90.42}},
+        {0.0025, {1.0, 1.1, 1.32, 1.61, 2.69, 4.48, 7.53, 13.65, 23.86, 45.66, 90.42}},
+        {0.0125, {1.0, 1.1, 1.32, 1.61, 2.69, 4.48, 7.53, 13.65, 23.86, 45.66, 90.42}},
+        {0.0625, {1.0, 1.1, 1.32, 1.61, 2.69, 4.48, 7.53, 13.65, 23.86, 45.66, 90.42}},
+        {0.3125, {1.0, 1.1, 1.32, 1.61, 2.69, 4.48, 7.53, 13.65, 23.86, 45.66, 90.42}}
     };
 
+    static std::map<double, std::array<double, 11>>::const_iterator find_closest_fpr(double const fpr)
+    {
+        auto lower_it = cost_factor.lower_bound(fpr);
+        auto upper_it = cost_factor.upper_bound(fpr);
+
+        assert(lower_it != cost_factor.end() || upper_it != cost_factor.end());
+
+        if (lower_it == cost_factor.end())
+            return upper_it;
+
+        if (upper_it == cost_factor.end())
+            return lower_it;
+
+        if (std::abs(lower_it->first - fpr) < std::abs(upper_it->first - fpr))
+            return lower_it;
+        else
+            return upper_it;
+    }
```

Comment:
```suggestion
    static std::array<double, 11> const & closest_factor(double const fpr)
    {
        if (auto it = cost_factors.find(fpr); it != cost_factors.end())
            return it->second;

        auto lower_it = cost_factors.lower_bound(fpr);
        auto upper_it = cost_factors.upper_bound(fpr);

        assert(lower_it != cost_factors.end() || upper_it != cost_factors.end());

        if (lower_it == cost_factors.end())
            return upper_it->second;

        if (upper_it == cost_factors.end())
            return lower_it->second;

        if (std::abs(lower_it->first - fpr) < std::abs(upper_it->first - fpr))
            return lower_it->second;
        else
            return upper_it->second;
    }
```

and then in exact/interpolated: `auto cost_factor = closest_factor(fpr)`